app/components/table.tsx

```diff
@@ -9,10 +9,49 @@ export function Table({
   children: React.ReactNode;
   className?: string;
 }) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isOverflowing, setIsOverflowing] = useState(false);
+
+  useEffect(() => {
```

Comment:
Can you please abstract this into its own hook called `useTableIsOverflowing` which returns the value?